### PR TITLE
add updatedb support for updating with newest log files

### PR DESCRIPTION
**Pull request by mitechie**
- Added checks to not re-download log files from swift that are already available locally
- Added a check to not attempt to reload a logfile into the db if it's already been loaded.
- Add new updatedb targets to the longrunning script and the makefile so that you can trigger an update vs a fresh setup. 
- Add readme documentation on ^

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,16 +15,32 @@ need credentials to the charmstore logs via swift. These can be obtained by
...
 
 
-Steps to get the system setup for reporting:
+Getting setup
+~~~~
 
   make sysdeps
   make
   source ~/path/to/novarc/for/swift
+  source .venv/bin/activate
```
Comment (mitechie) [inline README.md:24]: You need this to get the get.sh to use the correct locally venv swift client.

File: long-running/workloads.py [hunk 2]
```diff
@@ -39,9 +39,9 @@ def find_app(l):
 def main():
 
     for g in logs:
-        print "Found logs {0}".format(len(g))
```
Comment (mitechie) [inline long-running/workloads.py:42]: updated this script so that make lint passes

File: long-running/longrunning.py [hunk 3]
```diff
@@ -65,6 +65,51 @@ def recreate_db():
...
+            print("Processing: {0}".format(path))
+            path_parts = path.split('/')
+            logname = os.path.basename(path)
+
+            datestr = logname.\
+                replace('api.jujucharms.com.log-', '').\
+                replace('.anon.gz', '')
+
+            c = conn.cursor()
+            res = c.execute('''
```
Comment (mitechie) [inline long-running/longrunning.py:83]: This is the new bit here. This was pulled out to be shared between init and update commands. This bit here does the check if the log file's been loaded before.

**Comment by mitechie** [comment]
QA:

Run through the updated makefile steps for the updating logs section.

File: long-running/longrunning.py (hunk 3 above)
Comment (bac) [reply long-running/longrunning.py:83]: lint:
`long-running/longrunning.py:86:54: E231 missing whitespace after ','`

**Comment by bac** [comment]
# QA
`make updated` failed with:

```
Processing: logs/api/2/api.jujucharms.com.log-20170105.anon.gz
Traceback (most recent call last):
  File "long-running/longrunning.py", line 390, in <module>
    cli()
  File "/home/bac/git/gypsy-danger/.venv/lib/python3.5/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/bac/git/gypsy-danger/.venv/lib/python3.5/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/bac/git/gypsy-danger/.venv/lib/python3.5/site-packages/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/bac/git/gypsy-danger/.venv/lib/python3.5/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/bac/git/gypsy-danger/.venv/lib/python3.5/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "long-running/longrunning.py", line 331, in updatedb
    load_logfiles()
  File "long-running/longrunning.py", line 86, in load_logfiles
    ''', ['/'.join([path_parts[-2], logname]),])
sqlite3.InterfaceError: Error binding parameter 0 - probably unsupported type.
Makefile:74: recipe for target 'updatedb' failed
```

I ran it again and it succeeded.

File: long-running/longrunning.py [hunk 4]
```diff
@@ -65,6 +65,51 @@ def recreate_db():
...
+    for g in logs:
+        print("Found logs {0}".format(len(g)))
+        for path in g:
+            print("Processing: {0}".format(path))
+            path_parts = path.split('/')
+            logname = os.path.basename(path)
+
+            datestr = logname.\
+                replace('api.jujucharms.com.log-', '').\
+                replace('.anon.gz', '')
```
Comment (bac) [inline long-running/longrunning.py:80]: regex coward!

**Review by bac — APPROVED** [review]
Please look into the data problem with that one file and make lint pass.  Thanks @mitechie

**Comment by mitechie** [comment]
I can't tell what would cause the query to fail that way. The only argument is a '/'.join() which should always return a string type. 

sorry for the lint. had the fix locally and forgot to push. Updated.

**Pull request merged**